diff -ru libxml2-2.7.8/configure libxml2-2.7.8+versioning/configure
--- libxml2-2.7.8/configure	2010-11-04 13:28:14.000000000 -0400
+++ libxml2-2.7.8+versioning/configure	2012-02-15 09:21:41.959278546 -0500
@@ -11414,7 +11427,7 @@
   esac
 fi
 
- if test -z "$VERSION_SCRIPT_FLAGS"; then
+ if test -n "$VERSION_SCRIPT_FLAGS"; then
   USE_VERSION_SCRIPT_TRUE=
   USE_VERSION_SCRIPT_FALSE='#'
 else
diff -ru libxml2-2.7.8/configure.in libxml2-2.7.8+versioning/configure.in
--- libxml2-2.7.8/configure.in	2010-11-04 13:01:19.000000000 -0400
+++ libxml2-2.7.8+versioning/configure.in	2012-02-15 09:21:31.711278392 -0500
@@ -84,7 +84,7 @@
   esac
 fi
 AC_SUBST(VERSION_SCRIPT_FLAGS)
-AM_CONDITIONAL([USE_VERSION_SCRIPT], [test -z "$VERSION_SCRIPT_FLAGS"])
+AM_CONDITIONAL([USE_VERSION_SCRIPT], [test -n "$VERSION_SCRIPT_FLAGS"])
 
 dnl
 dnl We process the AC_ARG_WITH first so that later we can modify
