diff -up pulseaudio-0.9.21/po/POTFILES.in.potfiles pulseaudio-0.9.21/po/POTFILES.in
--- pulseaudio-0.9.21/po/POTFILES.in.potfiles	2010-12-08 11:16:57.386803004 -0500
+++ pulseaudio-0.9.21/po/POTFILES.in	2010-12-08 11:17:50.350803004 -0500
@@ -153,6 +153,7 @@ src/daemon/dumpmodules.c
 src/daemon/daemon-conf.c
 src/daemon/caps.c
 src/daemon/pulseaudio.desktop.in
+src/daemon/pulseaudio-kde.desktop.in
 src/pulse/channelmap.c
 src/pulse/error.c
 src/pulse/proplist.c
